.github/workflows/test.yml

```diff
@@ -0,0 +1,23 @@
+name: Test Hugo Build
+
+on:
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-22.04
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true  # Fetch Hugo themes (true OR recursive)
+          fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod
+
+      - name: Setup Hugo
```

Comment:
Ideally, this and the build used in `gh-pages.yml` should use the same Hugo setup to be 100% sure it actually builds if the test passes... 🤔

Reply:
I _think_ I set this up so that `bulid` runs on every PR, but `deploy` only on merges to master. Of course we can't test that until we merge it 🤞🏼 :)

Reply:
Ok, I think I misread on my first review. If I get this right you want to have two actions, one runs on every PR building the page and thus verifying it works, and the other deploying it on merge.

However, I think it won't work to split the build and the deploy job. The deploy job requires the output of the build job, but if they are run in different actions they are run in different workers - which means somewhat different "machines"...

I tried the action on my fork of the tw.org repository and got 
```
The workflow is not valid. .github/workflows/gh-pages.yml (Line: 30, Col: 3): The workflow must contain at least one job with no dependencies.
```
I think we will have to rebuild the page on merge, even if we already had built it in the PR... 🤔

Reply:
I think there's a `workflow_run` dependency, too, but maybe it's not possible to only run that dependency on pull requests. I'll just duplicate the `build` back into the deploy file.